src/frontends/onnx/onnx_common/src/onnx_model_validator.cpp

```diff
@@ -165,7 +165,7 @@ bool is_valid_model(std::istream& model) {
     const size_t EXPECTED_FIELDS_FOUND = 3u;
     std::unordered_set<::onnx::Field, std::hash<int>> onnx_fields_found = {};
     try {
-        while (!model.eof() && onnx_fields_found.size() < EXPECTED_FIELDS_FOUND) {
+        while (model && onnx_fields_found.size() < EXPECTED_FIELDS_FOUND) {
```

Comment:
not sure about it. @gkrivor, please take a look

Reply:
Also was not sure but after reading suggestion and docs it became more clear: checking `!model.eof()` when `model` is `std::stream` might be not enough to exit the while loop. E.g., in case of data reading failure, all subsequent calls to `eof` function will return false, which may lead to an infinite loop. At the same time `bool` operator for `std::stream` checks `model.fail()`: https://en.cppreference.com/w/cpp/io/basic_ios/operator_bool

Reply:
@gkrivor Any conclusion?

Reply:
@evgenytalanin-intel sorry for the delay :(

Looks like we have to solve two issues: first is encountered by the eof, second - by unbreakable loop in case of a streaming case.

Regarding latest documention about istream I may conclude eof is a deprecated feature :)  I suggest to replace "!model.eof()" it by "model.good()", regarding description here https://en.cppreference.com/w/cpp/io/basic_ios/good it returns true only in case everything is ok. as you mentioned - eof() returns an expected "false" only in case we've reached end of stream and ignores all other possible failures.

Because we are rewinding each received stream to its beginning (https://github.com/openvinotoolkit/openvino/blob/fe2f67bc7adfd122ece8d9be992657b6cdb0ff89/src/frontends/onnx/frontend/src/frontend.cpp#L175), we may use model.tellg() to limit reading by some pre-defined value.

I may suggest to limit reading by 1Mb. Don't know, I think it must be enough:
https://github.com/onnx/onnx/blob/2b504f81f1cadb098e69b53b76e4e12b66a3bdbf/onnx/onnx.proto3#L361

It describes an expected "ModelProto" and regarding this description first fields, in case of correct model, must be in a first 10k.

My suggestion is an adding condition:
&& model.tellg() < 1024*1024

Reply:
I agree with 1MB should be enough from common sense perspective but let's not introduce any magic constants. Changed bool operator to `.good`. Any other suggestions?